string place
---
string transition
---
int32[] time_passed
